invenio_workflows/worker_engine.py

```diff
@@ -38,17 +40,24 @@ def run_worker(wname, data, **kwargs):
     :param data: objects to run through the workflow.
     :type data: list
 
+    :param engine_uuid_hex: hex string of the uuid of the engine to use, if
+        not passed will create a new one.
+    :type data: str
+
     :return: WorkflowEngine instance
     """
     if 'stop_on_halt' not in kwargs:
         kwargs['stop_on_halt'] = False
 
-    engine = WorkflowEngine.with_name(wname, **kwargs)
-    engine.save()
-    objects = get_workflow_object_instances(data, engine)
+    if engine_uuid_hex:
+        engine_uuid = uuid.UUID(hex=engine_uuid_hex)
```

Comment:
This is run by the 'start' task (https://github.com/inveniosoftware-contrib/invenio-workflows/blob/master/invenio_workflows/tasks.py#L31), and that one just passed through `**kwargs` so I did not want to put there a special case for the engine_uuid (getting the string and creating the UUID to pass it to this `run_worker`), so the only other place to do the transformation is here.